### DISCUSSION TITLE
detect router change and stop it

### DISCUSSION THREAD

**Discussion by MateoKappa** [vercel/next.js#54144] [App Router]
So im trying to create a "you have unsaved changes are you sure you want to leave" event in my app , im using next js 13 app dir layout and since react events are not available at this , i cannot find an alternative does someone has a suggestion ? Also beforeunload event doesnt work , it only works when im about to close the page complitely but not when im changing routes in my web app

**Comment by moroshko** [comment]
@delbaoliveira This is a very common use case. What the current recommendation on this?
